From eb8b3f27e2937d8856fe01441b6c2880b59f02b5 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Sun, 10 Jan 2016 12:52:15 +0800
Subject: [PATCH] MLK-12200 common: cli_simple: use strlcpy instead of strcpy

Report Coverity log:
Destination buffer too small (STRING_OVERFLOW)
string_overflow: You might overrun the 1024 byte destination string
lastcommand by writing 1025 bytes from console_buffer

Signed-off-by: Peng Fan <peng.fan@nxp.com>
---
 common/cli_simple.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common/cli_simple.c b/common/cli_simple.c
index 6c65cc6..690f00b 100644
--- a/common/cli_simple.c
+++ b/common/cli_simple.c
@@ -275,7 +275,8 @@ void cli_simple_loop(void)
 
 		flag = 0;	/* assume no special flags for now */
 		if (len > 0)
-			strcpy(lastcommand, console_buffer);
+			strlcpy(lastcommand, console_buffer,
+				CONFIG_SYS_CBSIZE + 1);
 		else if (len == 0)
 			flag |= CMD_FLAG_REPEAT;
 #ifdef CONFIG_BOOT_RETRY_TIME
-- 
1.8.0

